src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -14,14 +14,14 @@ namespace Microsoft.AspNetCore.Razor.Runtime.TagHelpers
     /// </summary>
     public class TagHelperExecutionContext
     {
-        private readonly string _tagName;
-        private readonly string _uniqueId;
-        private readonly TagMode _tagMode;
         private readonly List<ITagHelper> _tagHelpers;
-        private readonly Func<Task> _executeChildContentAsync;
         private readonly Action<HtmlEncoder> _startTagHelperWritingScope;
         private readonly Func<TagHelperContent> _endTagHelperWritingScope;
```

Comment:
Has always been strange that these were passed into `TagHelperScopeManager.Begin()`. MVC doesn't have a reason to change the values from call to call. (Don't recall if it ever passed anything but a method name.) So, while we could pile more into `Reinitialize()`, better to give the values a sensible lifetime.

Bottom line, change the `TagHelperScopeManager` constructor or file a follow-up issue to do so.